File: js/task3a/S3_OBJ_03.js
function pick(obj, keys) {
  return keys.reduce((res, key) => {
    if (Object.prototype.hasOwnProperty.call(obj, key)) {
      res[key] = obj[key];
    }
    return res;
  }, {});
}

const obj = { a: 1, b: 2, c: 3 };
console.log(pick(obj, ['a'])); 
console.log(pick(obj, ['a', 'c'])); 
console.log(pick(obj, ['b', 'd'])); 
console.log(pick(obj, [])); 
console.log(pick(obj, ['a', 'b', 'c', 'd'])); 

File: python/task3a/S3_OBJ_03.py
def pick(obj, keys):
  return {k: obj[k] for k in keys if k in obj}


obj = {"a": 1, "b": 2, "c": 3}

print(pick(obj, ["a"]))
print(pick(obj, ["a", "c"]))
print(pick(obj, ["b", "d"]))
print(pick(obj, []))
print(pick(obj, ["a", "b", "c", "d"]))
